app/controllers/ratings_controller.rb

```diff
@@ -0,0 +1,27 @@
+class RatingsController < ApplicationController
+  before_action :get_shopping_cart
+
+  def rate
```

Comment:
i think update would be more appropriate in that case because we are not creating a new rating only updating it.